packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -280,7 +286,9 @@ async function nodeImport(
       typeof jest === 'undefined'
         ? { ...resolveOptions, tryEsmOnly: true }
         : resolveOptions,
-      false,
+      targetWeb,
+      undefined,
+      true,
```

Comment:
```suggestion
      false,
      undefined,
      true,
```
Because this part imitates Node and Node's resolver works like `targetWeb=false`, this needs to be `false`.
If we run this part with `true`, the same code won't run after build because of that.

Reply:
Hmm, but afaik a main use case for `ssr.target = 'webworker'` is for when the end user is planning to run in non-node runtimes (cloudflare, deno, bun, etc). This is how I've been using it at least - for example when using `ssrLoadModule` in dev to load a file that imports `renderToReadableStream` from `react-dom`.

Although I suppose the end user can now achieve this same result by explicitly listing the relevant conditions in `ssr.resolve` - the inconsistency between how dev server pipeline and build pipeline use ssr.target just felt a little weird.

Since there's another way to achieve the desired result via the new ssr.resolve options, I can change it back to false, just please confirm with a 👍 when you have a sec.

Reply:
> Hmm, but afaik a main use case for `ssr.target = 'webworker'` is for when the end user is planning to run in non-node runtimes (cloudflare, deno, bun, etc).

If a user is going to create a bundle for non-node runtimes, we expect them to set `ssr.noExternal: true`.
https://vitejs.dev/guide/ssr.html#ssr-bundle
This is because it doesn't make sense to externalize any modules in that case. (Bare import specifiers won't work in those environments because those environments doesn't support `node_modules` resolutions.)
If the module is not externalized, `ssr.target = 'webworker'` would set the `targetWeb` for vite:resolve plugin and that would resolve the import instead of `ssrLoadModule`.

Reply:
Makes sense, but I still think there are some rough edges when trying to use vite createServer + ssrLoadModule to create a nice local dev server experience (in conjunction w actually running dev server in non node runtimes like bun, or when wanting to better represent the final build target during local dev when that build target is non-node like cloudflare). BUT I think best as a separate discussion, the adjustments in this PR make it possible to cover everything :). 

I reverted the targetWeb change - thanks for the review!